Framework/MDAlgorithms/src/MDTransfModQ.cpp

```diff
@@ -166,22 +166,28 @@ the algorithm and false otherwise.
 * it also uses preprocessed detectors positions, which are calculated by
 PreprocessDetectors algorithm and set up by
 * calcYDepCoordinates(std::vector<coord_t> &Coord,size_t i) method.    */
-bool MDTransfModQ::calcMatrixCoordInelastic(const double &E_tr,
+bool MDTransfModQ::calcMatrixCoordInelastic(const double &deltaE,
                                             std::vector<coord_t> &Coord) const {
-  if (E_tr < m_DimMin[1] || E_tr >= m_DimMax[1])
+  if (deltaE < m_DimMin[1] || deltaE >= m_DimMax[1])
     return false;
-  Coord[1] = static_cast<coord_t>(E_tr);
-  double k_tr;
-  // get module of the wavevector for scattered neutrons
+  Coord[1] = static_cast<coord_t>(deltaE);
+
+  // x,y,z refer to internal coordinate system where Z is the beam direction
+  double qx{0.0}, qy{0.0}, qz{0.0};
   if (this->m_Emode == Kernel::DeltaEMode::Direct) {
-    k_tr = sqrt((m_Ei - E_tr) / PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    const double kFinal = sqrt((m_eFixed - deltaE) /
+                               PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    qx = -m_ex * kFinal;
+    qy = -m_ey * kFinal;
+    qz = m_kFixed - m_ez * kFinal;
   } else {
-    k_tr = sqrt((m_Ei + E_tr) / PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    const double kInitial = sqrt(
```

Comment:
@AndreiSavici Can you describe exactly why you think this is wrong? It doesn't just flip the sign from before we think the old formula was just wrong for Indirect. The variable naming meant that `k_i` got set to `kfixed` for indirect. Maybe BASIS operates in a regime where the differences are less prominent, e.g. lower energy transfer?

We've had 3 people derive it separately and come to this answer.